### PR TITLE
minor refactor: move routeName method to the Route

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message: minor refactor: move routeName method to the Route
Additional Description: 

This PR did two things:
1) Move the `routeName()` method from `DirectResponseEntry` and `RouteEntry` to Route because this attribute belongs to single route rather than belongs to one of route actions.
2) Avoid setting route name to stream info because we have set route into the stream info. We can get route name from route directly.

Risk Level: n/a.
Testing: unit?
Docs Changes: n/a.
Release Notes: n/a.
Platform Specific Features: n/a.

### CODE REVIEW

**Comment by wbpcode** [comment]
This PR is used to improve code quality. But it will also brings minor optimization to memory overhead (32+bytes for every stream info) and performance (one time memory copy and even one time memory allocation). 😄

**Comment by wbpcode** [comment]
/wait 

will fix ci tomorrow

**Comment by wbpcode** [comment]
ready for a formal review.

**Comment by htuch** [comment]
@KBaichoo want to take main pass?

**Review by KBaichoo — APPROVED** [review]
Nice!

/retest

**Comment by wbpcode** [comment]
main was broken, wait main to be fixed first.

**Comment by wbpcode** [comment]
friendly ping @htuch for a second review or merge.

**Review by htuch — APPROVED** [review]

File: source/common/stream_info/stream_info_impl.h [hunk 1]
```diff
@@ -238,12 +240,10 @@ struct StreamInfoImpl : public StreamInfo {
 
   uint64_t responseFlags() const override { return response_flags_; }
 
-  void setRouteName(absl::string_view route_name) override {
-    route_name_ = std::string(route_name);
+  const std::string& getRouteName() const override {
+    return route_ != nullptr ? route_->routeName() : EMPTY_STRING;
```
Comment (kyessenov) [inline source/common/stream_info/stream_info_impl.h:244]: @wbpcode I think the route_ is never set for direct response action? It looks like it's only set in the upstream request which is not happening for direct response.

Comment (wbpcode) [reply source/common/stream_info/stream_info_impl.h:244]: Nope, the route_ will be set when we refresh cached route. See

image

Comment (wbpcode) [reply source/common/stream_info/stream_info_impl.h:244]: did you encounter some problem in accutally usage?

Comment (zirain) [reply source/common/stream_info/stream_info_impl.h:244]: @wbpcode it broken istio's CI, see https://github.com/istio/proxy/pull/4932.

Comment (wbpcode) [reply source/common/stream_info/stream_info_impl.h:244]: Could you provide more speicifc log or problem or context? It's a little hard for me to figure out the root cause or problem.

Comment (zirain) [reply source/common/stream_info/stream_info_impl.h:244]: It breaks the implementation of stackdrive filter, @kyessenov can you share the details?

Comment (kyessenov) [reply source/common/stream_info/stream_info_impl.h:244]: I think this PR fixed some bug actually where `route_name` was not populated with RBAC. Now it's populated, and we can update our log entry samples. Not sure if this deserves a notice - it's improving telemetry overall.

**Pull request merged**